src/buffer.rs

```diff
@@ -145,6 +152,12 @@ impl<'a, P: Pixel + 'a> Iterator for PixelsMut<'a, P> where P::Subpixel: 'a {
     }
 }
 
+impl <'a, P: Pixel + 'a> ExactSizeIterator for PixelsMut<'a, P> where P::Subpixel: 'a {
```

Comment:
Can the entire where clause and all bounds on `P` be replaced with `where Self: Iterator`?